### PR TITLE
test: add integration tests and update deps

### PR DESCRIPTION
**Pull request by arturovt**
@NetanelBasal FYI

- updated dependencies
- added **real** integration tests that run with the Ivy engine

### CODE REVIEW

File: integration/app/app.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      template: '<div test></div>'
+    })
+    class MockComponent {}
+
+    @UntilDestroy({ checkProperties: true })
+    @Directive({ selector: '[test]' })
+    class MockDirective {
+      disposed = false;
+
+      subscription = of(10)
```
Comment (NetanelBasal) [inline integration/app/app.component.spec.ts]: `of` isn't the "best" example as it fires once and completes.

Comment (arturovt) [reply integration/app/app.component.spec.ts]: Y, I forgot to change to subject 😝 
Will do

Comment (arturovt) [reply integration/app/app.component.spec.ts]: Done.

Comment (arturovt) [reply integration/app/app.component.spec.ts]: Forgot to add a test for `untilDestroyed`.

**Comment by NetanelBasal** [comment]
LMK when you added the `untilDestroyed` spec, and I will merge. Thanks.

**Comment by arturovt** [comment]
> LMK when you added the `untilDestroyed` spec, and I will merge. Thanks.

Done.

**Pull request merged**